### ISSUE TITLE
Parameterize CLB configuration with validation options

### ISSUE THREAD

**Issue by uhbif19** [mlabs-haskell/clb#24]
Usable options I see:

- [ ] protocolVersion (some changes in Plutus (and ledger?) behavior is linked to it, for example [this bug fix](https://github.com/IntersectMBO/cardano-ledger/issues/3043))
- [ ] There are some datatype for disabling validations (like `ValidateAll`), but I have not found external toggle for it so far

**Comment by euonymos** [comment]
Also - don't use cryptography (if it's possible).

**Comment by uhbif19** [comment]
@euonymos 

More info on options.

There is [config](https://github.com/IntersectMBO/cardano-ledger/blob/c2fd86ea034a8021766a794cd06f888b9cfbf62c/libs/small-steps/src/Control/State/Transition/Extended.hs
) to disable validation for any ledger rules. But I am not sure how to apply it in our case.

As for feature of fee disabling discussed on last meeting: seems like there is at least two ways to do it, this one, and making fees zero.

Also there are some "events" some of [which may cover](https://github.com/IntersectMBO/cardano-ledger/blob/ab8d57cf43be912a336e872b68d1a2526c93dc6a/docs/LedgerEvents.md
) some of our needs.

**Comment by euonymos** [comment]
No, we use an instance of `Cardano.Ledger.Crypto.Crypto`, which is [`StandardCrypto`](https://github.com/IntersectMBO/cardano-ledger/blob/master/libs/cardano-ledger-core/src/Cardano/Ledger/Crypto.hs#L40-L45). By using mock hashes we can probably speed everything up a bit. Take a look [here](https://github.com/IntersectMBO/cardano-ledger/blob/master/eras/shelley/test-suite/src/Test/Cardano/Ledger/Shelley/ConcreteCryptoTypes.hs#L52-L57).
`

**Comment by euonymos** [comment]
Disabling ledger predicates might be an issue. Adding any instrumentation to the ledger is not an option (and we don't want to maintain our fork). AFAIK there is no way to "turn off" any rules, we have to find ways to circumvent them in other ways, like using parameters.

**Comment by euonymos** [comment]
> There is [config](https://github.com/IntersectMBO/cardano-ledger/blob/c2fd86ea034a8021766a794cd06f888b9cfbf62c/libs/small-steps/src/Control/State/Transition/Extended.hs) to disable validation for any ledger rules.

Do you mean [this](https://github.com/IntersectMBO/cardano-ledger/blob/c2fd86ea034a8021766a794cd06f888b9cfbf62c/libs/small-steps/src/Control/State/Transition/Extended.hs#L337-L343)?

**Comment by uhbif19** [comment]
I mean `ValidateSuchThat ([Label] -> Bool)` constructor for `ValidationPolicy`. It is passed inside `applyTx`, so we can use any. It seems like exactly what we need, if works correctly.

There are two `Label`s in this file so I am not sure which one :D Second is just string, so I assumed only rule as whole can be disabled.

**Comment by euonymos** [comment]
> ValidationPolicy



> I mean `ValidateSuchThat ([Label] -> Bool)` constructor for `ValidationPolicy`. It is passed inside `applyTx`, so we can use any. It seems like exactly what we need, if works correctly.

Sorry, I see it's passed for `applyBlock` and `applyTMiick`, but not for `applyTx`. (Cannot check the latest ledger's code though right now, using what I have locally).

**Comment by uhbif19** [comment]
@euonymos 

Sure. That is why we need to inline it: https://github.com/mlabs-haskell/clb/pull/33/commits

**Comment by uhbif19** [comment]
My initial approach did not work, using `ValidateSuchThat` did not seem to change behavior.

So I tried to parse `L.ApplyTxError` to get kind of error and retry `sendTx` in that case, but I failed to win its pattern matching, all its entries look like `LS.UtxowFailure (LB.UtxoFailure  (LB.AlonzoInBabbageUtxoPredFailure (LA.UtxosFailure (LS.ExpiredUTxO {}))))` (where `LS`/`LB`... are different versions of ledgers).
